viewing_party/party.py

```diff
@@ -1,23 +1,180 @@
 # ------------- WAVE 1 --------------------
 
+from __future__ import unicode_literals
+
+
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+    movie = {
+        "title": title,
+        "genre": genre,
+        "rating": rating,
+    }
+    return movie
+def add_to_watched(user_data, movie):
+    updated_data = user_data.copy()
+    updated_data["watched"].append(movie)
+    return updated_data
+def add_to_watchlist(user_data, movie):
+    updated_data = user_data.copy()
+    updated_data["watchlist"].append(movie)
+    return updated_data
+def watch_movie(user_data, title):
+   
+
+    watchlist = user_data["watchlist"] 
+    watched =user_data["watched"]
+   
+
+    for movie in watchlist:
+        if movie["title"] == title:
+            watched.append(movie)
+            watchlist.remove(movie)
+             
+    
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    # represents that the user has a list of watched movies 
+    if not user_data["watched"]:
+        return 0.0

```

Comment:
Nice use of a guard clause here by putting the if statement at the beginning of your function and then immediately returning 0 without needing to run the rest of the logic in your function. Line 44 is very pythonic (`if not user_data["watched"]`) since an empty list is Falsey.